internal/compiler/compiler.go

```diff
@@ -192,7 +192,11 @@ func (c *Compiler) getSpecialVars(t *ast.Task, call *ast.Call) (map[string]strin
 		"TASK_VERSION":     version.GetVersion(),
 	}
 	if t != nil {
-		maps.Copy(allVars, map[string]string{"TASK": t.Task, "TASKFILE": t.Location.Taskfile, "TASKFILE_DIR": filepath.Dir(t.Location.Taskfile)})
+		TaskDir := t.Dir
+		if !filepath.IsAbs(t.Dir) {
+			TaskDir = filepathext.SmartJoin(c.Dir, t.Dir)
+		}
+		maps.Copy(allVars, map[string]string{"TASK": t.Task, "TASKFILE": t.Location.Taskfile, "TASKFILE_DIR": filepath.Dir(t.Location.Taskfile), "TASK_DIR": TaskDir})
 	}
 	if call != nil {
 		maps.Copy(allVars, map[string]string{"ALIAS": call.Task})
```

Comment:
- No real reason to use `maps.Copy` here over manually assigning. I think doing it manually scales better in terms of readability.
- `filepathext.SmartJoin` already does an Abs check. I don't think we need to do it twice.
- Also prefer camelCase for `taskDir`.

```suggestion
	if t != nil {
		allVars["TASK"] = t.Task
		allVars["TASK_DIR"] = filepathext.SmartJoin(c.Dir, t.Dir)
		allVars["TASKFILE"] = t.Location.Taskfile
		allVars["TASKFILE_DIR"] = filepath.Dir(t.Location.Taskfile)
	}
	if call != nil {
		allVars["ALIAS"] = call.Task
	}
```